File: feishu-print-backend/app/models/signature.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Enum
from sqlalchemy.sql import func
from ..database import Base
import enum

class SignatureStatus(str, enum.Enum):
    pending = "pending"
    signed = "signed"
    expired = "expired"

class Signature(Base):
    __tablename__ = "signatures"
    
    id = Column(Integer, primary_key=True, index=True)
    template_id = Column(Integer, nullable=True)
    record_id = Column(String(255), nullable=True)
    signer_name = Column(String(100), nullable=False)
    signer_email = Column(String(255), nullable=True)
    document_title = Column(String(255), nullable=True)
    document_html = Column(Text, nullable=True)
    signature_data = Column(Text, nullable=True)
    signed_at = Column(DateTime(timezone=True), nullable=True)
    status = Column(String(20), default="pending")
    token = Column(String(64), unique=True, index=True, nullable=False)
    expires_at = Column(DateTime(timezone=True), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: feishu-print/src/api/signature.ts
import axios from 'axios';

const API_BASE = import.meta.env.VITE_API_BASE || 'http://localhost:8000';

export interface SignatureRequest {
  template_id?: number;
  record_id?: string;
  signer_name: string;
  signer_email?: string;
  document_title?: string;
  document_html?: string;
  expires_hours?: number;
}

export interface SignatureData {
  id: number;
  template_id?: number;
  record_id?: string;
  signer_name: string;
  signer_email?: string;
  document_title?: string;
  document_html?: string;
  signature_data?: string;
  signed_at?: string;
  status: 'pending' | 'signed' | 'expired';
  token: string;
  expires_at: string;
  created_at: string;
}

export const signatureApi = {
  create: async (data: SignatureRequest): Promise<SignatureData> => {
    const response = await axios.post(`${API_BASE}/api/signatures/`, data);
    return response.data;
  },

  getByToken: async (token: string): Promise<SignatureData> => {
    const response = await axios.get(`${API_BASE}/api/signatures/token/${token}`);
    return response.data;
  },

  submit: async (token: string, signatureData: string): Promise<SignatureData> => {
    const response = await axios.put(`${API_BASE}/api/signatures/token/${token}`, {
      signature_data: signatureData
    });
    return response.data;
  },

  getById: async (id: number): Promise<SignatureData> => {
    const response = await axios.get(`${API_BASE}/api/signatures/${id}`);
    return response.data;
  },

  getAll: async (): Promise<SignatureData[]> => {
    const response = await axios.get(`${API_BASE}/api/signatures/`);
    return response.data;
  },

  generateLink: (token: string): string => {
    const baseUrl = window.location.origin;
    const baseDir = import.meta.env.BASE_URL.replace(/\/$/, '');
    return `${baseUrl}${baseDir}/sign/${token}`;
  }
};
